osu.Game.Rulesets.Osu/Objects/Spinner.cs

```diff
@@ -31,6 +33,15 @@ protected override void ApplyDefaultsToSelf(ControlPointInfo controlPointInfo, B
             SpinsRequired = (int)Math.Max(1, SpinsRequired * 0.6);
         }
 
+        protected override void CreateNestedHitObjects()
+        {
+            base.CreateNestedHitObjects();
+
+            var maximumSpins = OsuAutoGeneratorBase.SPIN_RADIUS * (Duration / 1000) / MathHelper.TwoPi;
```

Comment:
Is this basically a guess at the maximum number of ticks, or is this grounded in mathematical truth? I'm finding the 2pi value to be rather arbitrary here...

What if a user spins closer than the spin radius (which generally players that spin 477 do)? Is it possible for the user to out-spin the number of ticks?
At the moment the spinner isn't capped to a maximum speed (for better or worse) so it seems like this should be an issue.

I'm also wondering if we can add the ticks dynamically to avoid this issue, since they're counted as bonus as far as the score processor is concerned.

Reply:
fwiw you can test this with auto generation by changing the following two lines to be `/ 10` rather than `/ 20` in `OsuAutoGenerator.cs`:

![image](https://user-images.githubusercontent.com/1329837/71734280-e9257000-2e8e-11ea-9a97-d2888d4917e6.png)

Reply:
You know, I think this could be a pretty cool way of limiting sliders: removing the max rpm limitation but limiting the maximum bonus score. @smoogipoo thoughts?

Reply:
What do you achieve by doing that? In the first place I can't spin >=477 so I don't have the ability to determine what the maximum bonus points should be by myself.
The situation I'd worry about is underestimating that value and then causing someone who spins 430 to be valued the same as someone that spins 350. Or in lazer terms, someone spinning like 750 vs someone spinning 550.

Reply:
Just another step in the direction of removing complex interactions that really don't need to be there.

My thinking is that it will make calculations much simpler (and also simplify implementation of spinners as it would be 1:1 to input rather than artificially limited at a per-frame basis).

We've seen how annoying spinners can be with stable; part of that comes from the per-frame limiting. Even with our existing replay system we would be potentially losing data – input/update at 1000hz but recording far lower (obviously we could attach as extra data in the replay frames, but that makes things more complex once again).

If we were to remove the limiter, it is 1:1 all the way, and the SPM meter would also match near-perfectly where that would otherwise not be possible.

The value can/would of course be tweaked to ensure what you say doesn't happen.

Reply:
Sure, let's see how it behaves for now.